apm-agent-attach/src/main/java/co/elastic/apm/attach/ElasticApmAttacher.java

```diff
@@ -168,15 +156,29 @@ static String toAgentArgs(Map<String, String> configuration) {
      * @param configuration the agent configuration
      */
     public static void attach(String pid, Map<String, String> configuration) {
-        ByteBuddyAgent.attach(AgentJarFileHolder.INSTANCE.agentJarFile, pid, toAgentArgs(configuration), ATTACHMENT_PROVIDER);
+        // optimization, this is checked in AgentMain#init again
+        if (Boolean.getBoolean("ElasticApm.attached")) {
+            return;
+        }
+        File tempFile = createTempProperties(configuration);
+        String agentArgs = tempFile == null ? null : TEMP_PROPERTIES_FILE_KEY + "=" + tempFile.getAbsolutePath();
+
+        ByteBuddyAgent.attach(AgentJarFileHolder.INSTANCE.agentJarFile, pid, agentArgs, ATTACHMENT_PROVIDER);
+        if (tempFile != null) {
+            if (!tempFile.delete()) {
+                tempFile.deleteOnExit();
+            }
+        }
     }
 
     /**
      * Attaches the agent to a remote JVM
      *
      * @param pid       the PID of the JVM the agent should be attached on
      * @param agentArgs the agent arguments
+     * @deprecated use {@link #attach(String, Map)}
      */
+    @Deprecated
```

Comment:
Why deprecating and not removing?

Reply:
Because it's a public API

Reply:
With PID? It's only relevant to remote attach, isn't it?